adagrams/game.py

```diff
@@ -1,11 +1,193 @@
+    # make empty letter pool list
+    # iterate over dictionary to add each letter to the letter pool list however many times its value says to
+    # return letter_pool_list
+    letter_pool_list = []
+    for letter, frequency in LETTER_POOL.items():
+        for i in range(frequency):
+            letter_pool_list.append(letter)
+
+    return letter_pool_list
+
 def draw_letters():
-    pass
+    # use rand.randint(range) to get a random number
+    # use random number as index to pull letter from copy of pool of letters. 
+    # once we have 10 letters, return new list of single letters as hand
+    letter_pool_list = create_letter_pool_list(LETTER_POOL)
+    current_hand = []
+    for i in range(10):
+        pool_index = random.randint(0, len(letter_pool_list)-1)
+        draw_letter = letter_pool_list[pool_index]
+        current_hand.append(draw_letter)
+        letter_pool_list.remove(draw_letter)
+    return current_hand    
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # make copy of letter_bank that we can change
+    bank_copy = letter_bank.copy()
+    # loop through word
+    for letter in word.upper():
+        if letter not in bank_copy:
+            return False
+        else:
+            bank_copy.remove(letter)
+    return True
+
 
 def score_word(word):
-    pass
+    #create a dictionary with these point values
+    '''Create a dictionary score_board with key-value pairs 
+    that are letter and corresponding value.
+    Initialize the result to 0
+    Loop through the characters in the word
+    Check if the character is in the letter_vals dictionary
+    If it is, add the value of the letter to the result
+    Return the result'''
+
+    result = 0
+    extra_point = [7,8,9,10]
+    
+    values_dict = {
+        'A': 1, 
+        'B': 3, 
+        'C': 3, 
+        'D': 2, 
+        'E': 1, 
+        'F': 4, 
+        'G': 2, 
+        'H': 4, 
+        'I': 1, 
+        'J': 8, 
+        'K': 5, 
+        'L': 1, 
+        'M': 3, 
+        'N': 1, 
+        'O': 1, 
+        'P': 3, 
+        'Q': 10, 
+        'R': 1, 
+        'S': 1, 
+        'T': 1, 
+        'U': 1, 
+        'V': 4, 
+        'W': 4, 
+        'X': 8, 
+        'Y': 4, 
+        'Z': 10
+    }
+
+
+    for letter in word:
+        if letter.isalpha():
+            result += values_dict[letter.upper()]
+        else:
+            continue
+    if len(word) in extra_point:
+        result += 8
+    return result
+
+
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    Finds highest scoring word and returns a tuple with winning word and score
+    In case of a tie:
+    - 10 letter word wins
+    - if no 10 letter words, shortest word wins
+    - if two words of same length, whichever comes first in the word list wins
+
+    Parameters: a list of strings
+
+    Returns: a tuple with one string and an integer
+    """
+    word_score_list = []
+    highest_scoring_words = []
```

Comment:
I recommend moving the declaration for this down to line 150, inside of the else-block so that we only allocate memory for the variable if it will definitely be used.